File: solutions/java/darts/1/src/main/java/Darts.java
class Darts {
    private double lastDistance = 0;
    private int score = 0;

    Darts(double x, double y) {
        this.scoreThrow(x, y);
    }

    void scoreThrow(double x, double y) {
        this.lastDistance = Math.sqrt(Math.pow(x, 2) + Math.pow(y, 2));
        if (this.lastDistance <= 1) {
            this.score += 10;
        } else if (this.lastDistance <= 5) {
            this.score += 5;
        } else if (this.lastDistance <= 10) {
            this.score += 1;
        }
    }

    int score() {
        return this.score;
    }

}

File: solutions/c/darts/1/src/darts.c
#include "darts.h"
#include "math.h"

uint8_t score(coordinate_t landing_position)
{
    float distance = sqrt(pow(landing_position[0], 2) + pow(landing_position[1], 2));

    if (distance > 10)
    {
        return 0;
    }
    else if (distance > 5)
    {
        return 1;
    }
    else if (distance > 1)
    {
        return 5;
    }
    else if (distance >= 0)
    {
        return 10;
    }
    else
    {
        return 0;
    }
}